tun/src/windows/mod.rs

```diff
@@ -15,15 +17,22 @@ pub struct TunInterface {
 impl TunInterface {
     #[throws]
     pub fn new() -> TunInterface {
-        let name = U16CString::from(u16cstr!("Burrow"));
+        Self::new_with_options(TunInterfaceOptions::new())?
+    }
+
+    #[throws]
+    pub(crate) fn new_with_options(options: TunInterfaceOptions) -> TunInterface {
+        let name_owned = options.name.unwrap_or("Burrow".to_owned());
```

Comment:
This seems good for now, but I am going to investigate what happens if we pass an empty name. It appears as if `wintun` might have handling for this